Add cvwsoc DTB to build targets.

Upstream-Status: Pending

diff --git a/arch/riscv/boot/dts/Makefile b/arch/riscv/boot/dts/Makefile
index fdae05bbf556..38d27a6310cf 100644
--- a/arch/riscv/boot/dts/Makefile
+++ b/arch/riscv/boot/dts/Makefile
@@ -8,4 +8,6 @@ subdir-y += sophgo
 subdir-y += starfive
 subdir-y += thead
 
+dtb-y += PLACEHOLDER.dtb
+
 obj-$(CONFIG_BUILTIN_DTB) := $(addsuffix .dtb.o, $(CONFIG_BUILTIN_DTB_SOURCE))
