Set max slicetime (useful for QEMU run only).

Upstream-Status: Pending

diff --git a/fbdoom/i_sound.c b/fbdoom/i_sound.c
index 5fee051..116ecff 100644
--- a/fbdoom/i_sound.c
+++ b/fbdoom/i_sound.c
@@ -46,7 +46,7 @@ int snd_cachesize = 64 * 1024 * 1024;
 // Config variable that controls the sound buffer size.
 // We default to 28ms (1000 / 35fps = 1 buffer per tic).
 
-int snd_maxslicetime_ms = 28;
+int snd_maxslicetime_ms = 100;
 
 // External command to invoke to play back music.
 
